CHANGES.md

```diff
@@ -1,6 +1,166 @@
 CHANGES
 =======
 
+4.0
+===
+
+This release contains several breaking changes. For a complete migration guide, see:
+https://django-csp.readthedocs.io/en/latest/migration-guide.html
+
+## Breaking Changes
+
+- **Configuration Format**: Moved to dict-based configuration which allows for setting policies for
+both enforced and report-only. Instead of using individual settings with `CSP_` prefixes, you now use
+dictionaries called `CONTENT_SECURITY_POLICY` and/or `CONTENT_SECURITY_POLICY_REPORT_ONLY`.
+
+  You can use Django's check command to automatically identify existing CSP settings and generate a
+  template for the new configuration format:
+
+  ```
+  python manage.py check
+  ```
+
+  This will detect your old `CSP_` prefixed settings and output a draft of the new dict-based
+  configuration, giving you a starting point for migration.
+
+  **Example:**
+
+  Change from:
+  ```python
+  CSP_DEFAULT_SRC = ["'self'", "*.example.com"]
+  CSP_SCRIPT_SRC = ["'self'", "js.cdn.com/example/"]
+  CSP_IMG_SRC = ["'self'", "data:", "example.com"]
+  CSP_EXCLUDE_URL_PREFIXES = ["/admin"]
+  ```
+
+  to:
+  ```python
+  from csp.constants import SELF
+
+  CONTENT_SECURITY_POLICY = {
+      "EXCLUDE_URL_PREFIXES": ["/admin"],
```

Comment:
Nit: for ease of reading, I would keep the order of the params the same as in the old version, above, so moving `EXCLUDE_URL_PREFIXES` to below `DIRECTIVES` (which also keeps it alpha-sorted)